### ISSUE TITLE
xmp:MetadataDate PDF/A validation fails

### ISSUE THREAD

**Issue by jbarlow83** [jbarlow83/OCRmyPDF#678]
https://www.pdf-online.com/osa/validate.aspx

Complains MetadataDate here, and not ModifyDate or other dates.

```xml
<xmp:ModifyDate>2020-11-14T10:58:55+00:00</xmp:ModifyDate>
<rdf:Description xmlns:xmp="http://ns.adobe.com/xap/1.0/" rdf:about="" xmp:MetadataDate="2020-11-14T02:58:55.648951"/></rdf:RDF>
```

So it seems to not approve of either fractional seconds, the lack of time zone, or the use of a date as an attribute instead of a value.

XMP spec 8.2.1.2:
https://wwwimages2.adobe.com/content/dam/acom/en/devnet/xmp/pdfs/XMP%20SDK%20Release%20cc-2016-08/XMPSpecificationPart1.pdf
It's the time zone.

At our stages, fix_docinfo.pdf is okay. metadata_fix.pdf introduces the error.

-> This is a pikedpf issue.

**Comment by jbarlow83** [comment]
Message is:
```
Validating file "_linn_no_ccs.pdf" for conformance level pdfa-2b

xmp:MetadataDate :: Wrong value type. Expected type 'Date'.

The document does not conform to the requested standard.

The document's meta data is either missing or inconsistent or corrupt.

The document does not conform to the PDF/A-2b standard.

Done.
```

**Comment by jbarlow83** [comment]
Fixed in pikepdf 2.1.2
